discovery/kubernetes/ingress.go

```diff
@@ -194,20 +195,18 @@ func (i *Ingress) buildIngress(ingress *v1beta1.Ingress) *targetgroup.Group {
 	}
 	tg.Labels = ingressLabels(ingress)
 
-	tlsHosts := make(map[string]struct{})
-	for _, tls := range ingress.Spec.TLS {
-		for _, host := range tls.Hosts {
-			tlsHosts[host] = struct{}{}
-		}
-	}
-
 	for _, rule := range ingress.Spec.Rules {
 		paths := pathsFromIngressRule(&rule.IngressRuleValue)
 
 		scheme := "http"
-		_, isTLS := tlsHosts[rule.Host]
-		if isTLS {
-			scheme = "https"
+	out:
```

Comment:
we have to walk this each time so it is more expensive than the original. I also thought we could maintain two maps, one for wildcard entries and one as before.